aten/src/THC/THCTensorRandom.cu

```diff
@@ -5,54 +5,45 @@
 #include <THC/THCTensorMath.h>
 #include <THC/THCReduceApplyUtils.cuh>
 #include <THC/THCTensorRandom.cuh>
-#include <THC/THCGenerator.hpp>
 #include <ATen/Config.h>
 
-#include <ATen/cuda/_curand_mtgp32_host.h>
-
 #include <thrust/functional.h>
 
 #define MAX_NUM_BLOCKS 200
 #define BLOCK_SIZE 256
 
-
-THCGenerator* THCRandom_getGenerator(THCState* state);
-
-/* Creates a new generator state given the seed. Not thread-safe. */
-__host__ void createGeneratorState(THCGenerator* gen, uint64_t seed)
+THC_API __host__ void THCRandom_getRNGState(at::Generator *gen_, THByteTensor *rng_state)
 {
-  // seed and offset for philox
-  gen->state.initial_seed = seed;
-  gen->state.philox_seed_offset = 0;
-}
-
-THC_API __host__ void THCRandom_getRNGState(THCState* state, THByteTensor *rng_state)
-{
-  THCGenerator* gen = THCRandom_getGenerator(state);
-  std::lock_guard<std::mutex> lock(gen->mutex);
-
-  // The RNG state comprises the seed, and an offset used for Philox
-  static const size_t states_size = MAX_NUM_BLOCKS * sizeof(curandStateMtgp32); // this line is just here for BC reason
-  static const size_t seed_size = sizeof(gen->state.initial_seed);
-  static const size_t offset_size = sizeof(gen->state.philox_seed_offset);
+  auto gen = at::check_generator<at::CUDAGenerator>(gen_);
+  std::lock_guard<std::mutex> lock(gen->mutex_);
+  // The RNG state comprises the seed, and an offset used for Philox.
+  // The following line is just here for BC reason. sizeof curandStateMtgp32 is 4120.
+  // It used to be static const size_t states_size = MAX_NUM_BLOCKS * sizeof(curandStateMtgp32);
+  // MAX_NUM_BLOCKS was 200 and sizeof(curandStateMtgp32) is 4120. Hardcoding these numbers here
+  // because this is just host side code and we don't want to worry about linking with cuda
+  static const size_t states_size = 200 * sizeof(4120);
+  static const size_t seed_size = sizeof(uint64_t);
+  static const size_t offset_size = sizeof(std::atomic<int64_t>);
```

Comment:
For funsies, did you know that the size of `std::atomic<int64_t>` is not guaranteed by the standard? :) :) :)

Reply:
Haha totally didn't know that std::atomic<int64_t> is not guaranteed.. in my defense we were doing this: `sizeof(gen->state.philox_seed_offset);` where `philox_seed_offset` was `std::atomic`. I'll change it to `sizeof(int64_t)` - I double checked that `sizeof(std::atomic<int64_t>)` isn't giving anything bigger than `sizeof(int64_t)` in g++ 5.4.0